src/clusterfuzz/_internal/build_management/build_archive.py

```diff
@@ -0,0 +1,331 @@
+
+    to_extract = []
+    for file in self.list_members():
+      filepath = file.name
+      path_components = os.path.normpath(filepath).split(os.sep)
+      # Is it an allowlisted binary?
+      if any(
+          component.startswith(allowlisted_names)
+          for component in path_components):
+        to_extract.append(file)
+        continue
+
+      basename = os.path.basename(filepath)
+      # Does it have a blocklisted extension?
+      if basename.endswith(blocklisted_extensions):
+        continue
+
+      # Does it have an extension?
+      if '.' in basename:
+        to_extract.append(file)
+
+    return to_extract
+
+  def list_fuzz_targets(self) -> List[str]:
+    if self._fuzz_targets:
+      return list(self._fuzz_targets.keys())
+    # Import here as this path is not available in App Engine context.
+    from clusterfuzz._internal.bot.fuzzers import utils as fuzzer_utils
+
+    for archive_file in self.list_members():
+      file_content = self.try_open(archive_file.name)
+      if fuzzer_utils.is_fuzz_target_local(archive_file.name, file_content):
+        fuzz_target = fuzzer_utils.normalize_target_name(archive_file.name)
+        self._fuzz_targets[fuzz_target] = archive_file.name
+
+      if file_content:
+        file_content.close()
+
+    return list(self._fuzz_targets.keys())
+
+  def unpacked_size(self, fuzz_target: Optional[str] = None) -> int:
+    if not fuzz_target:
+      return self.extracted_size()
+
+    files = self.get_target_dependencies(fuzz_target=fuzz_target)
+    return sum(f.size_bytes for f in files)
+
+  def unpack(self,
+             build_dir: str,
+             fuzz_target: Optional[str] = None,
+             trusted: bool = False) -> bool:
+    if not fuzz_target:
+      return self.extract_all(build_dir, trusted=trusted)
+    files = self.get_target_dependencies(fuzz_target=fuzz_target)
+    error_occured = False
+    for file in files:
+      error_occured |= self.extract(
+          file.name, build_dir, trusted=trusted) is None
+    return not error_occured
+
+
+class ChromeBuildArchive(DefaultBuildArchive):
+  """Handles chrome build archives. This special cases the default behaviour by
+  looking at the content of the `.runtime_deps` file, in order to unpack all the
+  fuzzer dependencies correctly.
+  In case something goes wrong, this defaults to using the default unpacker.
+  """
+
+  def root_dir(self) -> str:
+    if not hasattr(self, '_root_dir'):
+      self._root_dir = super().root_dir()
+    return self._root_dir
+
+  def to_archive_path(self, path: str) -> str:
+    """Deps are relative to the Chrome root directory. However, there might be
+    a common root directory in the archive, which means we need to make sure
+    the file path is correct.
+
+    Args:
+        path: the dependency path relative to Chrome's root directory.
+
+    Returns:
+        the path relative to the archive.
+    """
+    path = os.path.normpath(path)
+
+    if path.startswith('../../'):
```

Comment:
Does this work on windows where paths use backslashes?

Reply:
Yes, because GN always issue posix pathes. The tools are responsible for doing the conversion.